system/git_config.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Marius Gedminas <marius@pov.lt>
+# (c) 2016, Matthew Gamble <git@matthewgamble.net>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: git_config
+author:
+  - "Matthew Gamble"
+  - "Marius Gedminas"
+version_added: 2.1
+short_description: Read and write git configuration
+description:
+  - The M(git_config) module changes git configuration by invoking 'git config'.
+    This is needed if you don't want to use M(template) for the entire git
+    config file (e.g. because you need to change just C(user.email) in
+    /etc/.git/config).  Solutions involving M(command) are cumbersone or
+    don't work correctly in check mode.
+options:
+  list_all:
+    description:
+      - List all settings (optionally limited to a given scope)
+    required: false
+    default: no
+  name:
+    description:
+      - The name of the setting. If no value is supplied, the value will
+        be read from the config if it has been set.
+    required: false
+    default: null
+  repo:
+    description:
+      - Path to a git repository for reading and writing values from a
+        specific repo.
+    required: false
+    default: null
+  scope:
+    description:
+      - ...
+    required: false
+    default: null
```

Comment:
As this is a fixed list you can update the help to say which options are valid by adding:
`choices: [ 'local', 'global', 'system'] ]`